src/main.rs

```diff
@@ -46,26 +47,32 @@ enum Command {
     },
 }
 
-#[tokio::main]
-async fn main() -> Result<()> {
+fn main() -> Result<()> {
     tracing_subscriber::fmt::init();
 
-    let args = Args::parse();
-    match args.command {
-        Command::Local {
-            local_host,
-            local_port,
-            to,
-            port,
-            secret,
-        } => {
-            let client = Client::new(&local_host, local_port, &to, port, secret.as_deref()).await?;
-            client.listen().await?;
-        }
-        Command::Server { min_port, secret } => {
-            Server::new(min_port, secret.as_deref()).listen().await?;
-        }
-    }
+    let Args { command } = Args::try_parse()?;
 
-    Ok(())
+    Builder::new_multi_thread()
+        .enable_all()
+        .build()?
+        .block_on(async {
```

Comment:
Why not just use `macros` feature. Does it add a lot of overheard?

Reply:
My intent for separating this is so that we don't have to initialize the Tokio runtime in case CLI-parsing fails. Would you prefer this style instead? Note that the `async`-block syntax is functionally equivalent.

```rust
async fn run() -> anyhow::Result<> {
  // ...
}

fn main() -> anyhow::Result<()> {
  // ...
  Builder::new_multi_thread()
    .enable_all()
    .build()?
    .block_on(run())
}
```

Reply:
Or, perhaps, would you prefer to have it reverted altogether?

Reply:
This is equivalent to the following!

```rust
#[tokio::main]
async fn run() -> Result<()> {
  // ...
}

fn main() -> Result<()> {
  // ...
  run()
}
```

Reply:
Whoa! Today I learned... That's incredible! Applied your suggestion in the latest commit. 👍